web/src/main/java/com/epam/esm/controller/GiftCertificateController.java

```diff
@@ -1,59 +1,96 @@
 package com.epam.esm.controller;
 
-import com.epam.esm.dto.FilterRequest;
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.dto.mapper.impl.GiftCertificateMapperImpl;
 import com.epam.esm.entity.GiftCertificate;
-import com.epam.esm.exception.DaoException;
-import com.epam.esm.exceptions.IncorrectParameterException;
+import com.epam.esm.hateoas.impl.GiftCertificateHateoasAdder;
 import com.epam.esm.service.GiftCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.util.MultiValueMap;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("/gift_certificates")
 public class GiftCertificateController {
 
     private final GiftCertificateService giftCertificateService;
+    private final GiftCertificateMapperImpl mapper;
+    private final GiftCertificateHateoasAdder hateoasAdder;
 
-    public GiftCertificateController(GiftCertificateService giftCertificateService) {
+    @Autowired
+    public GiftCertificateController(GiftCertificateService giftCertificateService, GiftCertificateMapperImpl giftCertificateMapper, GiftCertificateHateoasAdder hateoasAdder) {
         this.giftCertificateService = giftCertificateService;
+        this.mapper = giftCertificateMapper;
+        this.hateoasAdder = hateoasAdder;
     }
 
-    @RequestMapping
-    public List<GiftCertificate> getAll() throws IncorrectParameterException, DaoException {
-        return giftCertificateService.getAll();
+    @GetMapping("/{id}")
```

Comment:
Please add swagger and configure it. And add information about controllers to swagger